flask_admin/templates/bootstrap3/admin/lib.html

```diff
@@ -179,9 +179,13 @@
 {% endmacro %}
 
 {% macro form_js() %}
-  {% if config.MAPBOX_MAP_ID %}
+  {% if config.MAPBOX_MAP_ID and config.MAPBOX_ACCESS_TOKEN %}
```

Comment:
This will prevent inclusion of the leaflet assets even for the v3 API, since MAPBOX_ACCESS_TOKEN isn't expected in the config in that case.